junk/gen-bench-data/src/main.rs

```diff
@@ -59,41 +76,64 @@ fn save_data(dest_dir: &Path, data: Data) -> Result<(), Box<dyn std::error::Erro
 
     create_dir_all(dest_dir)?;
 
-    save_file(dest_dir, "script.air", &data.air)?;
-    save_file(dest_dir, "prev_data.json", &reformat_json(&data.prev_data))?;
-    save_file(dest_dir, "cur_data.json", &reformat_json(&data.cur_data))?;
-    save_file(dest_dir, "params.json", &serde_json::to_vec_pretty(&data.params_json)?)?;
-    save_file(dest_dir, "keypair.ed25519", &data.keypair)?;
+    save_file(dest_dir, "script.air", Some(&data.air))?;
+    save_file(
+        dest_dir,
+        "prev_data.json",
+        reformat_json_if_possible(&data.prev_data),
+    )?;
+    save_file(
+        dest_dir,
+        "cur_data.json",
+        reformat_json_if_possible(&data.cur_data),
+    )?;
+    save_file(
+        dest_dir,
+        "params.json",
+        Some(&serde_json::to_vec_pretty(&data.params_json)?),
+    )?;
+    save_file(dest_dir, "keypair.ed25519", Some(&data.keypair))?;
 
     Ok(())
 }
 
 /// make zero-indentation data for better git diffs
-fn reformat_json(data: &[u8]) -> Vec<u8> {
+fn reformat_json_if_possible(data: &[u8]) -> Option<Vec<u8>> {
     use serde::ser::Serialize;
 
+    if data.is_empty() {
+        return None;
+    }
+
     let obj: serde_json::Value = serde_json::from_slice(data).unwrap();
     let fmt = serde_json::ser::PrettyFormatter::with_indent(&[]);
     let mut out = vec![];
     let mut ser = serde_json::ser::Serializer::with_formatter(&mut out, fmt);
     obj.serialize(&mut ser).unwrap();
-    out
+    Some(out)
 }
 
-fn save_file(dest_dir: &Path, filename: &str, data: impl AsRef<[u8]>) -> Result<(), Box<dyn std::error::Error>>{
+fn save_file(
+    dest_dir: &Path,
+    filename: &str,
+    data: Option<impl AsRef<[u8]>>,
+) -> Result<(), Box<dyn std::error::Error>> {
     use std::fs::*;
     use std::io::prelude::*;
 
     let mut dest_dir = dest_dir.to_owned();
     dest_dir.push(filename);
 
     let mut f = File::create(&dest_dir)?;
```

Comment:
Is it intended to create an empty file in case of no `data`? Otherwise it's better to first check for data presence and only then create a file.